fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java

```diff
@@ -2620,4 +2625,47 @@ public void markReadable() {
             assertTrue(indexer.asyncToSync(FDBStoreTimer.Waits.WAIT_ONLINE_BUILD_INDEX, indexer.markReadableIfBuilt()));
         }
     }
+
+    @Test
+    public void testConfigLoader() throws Exception {
+        final Index index = new Index("newIndex", field("num_value_unique"));
+        final RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.addIndex("MySimpleRecord", index);
+        };
+        openSimpleMetaData(hook);
+
+        try (FDBRecordContext context = openContext()) {
+            for (int i = 0; i < 1000; i++) {
+                TestRecords1Proto.MySimpleRecord record = TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(i).setNumValueUnique(i).build();
+                recordStore.saveRecord(record);
+            }
+            recordStore.clearAndMarkIndexWriteOnly(index).join();
+            context.commit();
+        }
+
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final CompletableFuture<Void> future;
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(index).setSubspace(subspace)
+                .setConfigLoader(old ->
+                        OnlineIndexer.Config.newBuilder()
```

Comment:
It would be nice to use `old.toBuilder()` instead and check things like`increaseLimitAfter` and `progressLogIntervalMillis` are kept.

Reply:
Oh, yeah. Good idea. Let me do that.